File: 2- First Year (2021-2022)/Term 1/Intro To Computing (CS211)/Lab/Week 11 - 2D Arrays/Task 1/Task 1.cpp
#include <iostream>
using namespace std;

int main()
{
	double rates[4][7], sum, avg;
	
	for (int i = 0; i < 4; i++) //weeks
	{
		sum = 0;
		avg = 0;
		cout << "Enter rainfall rate for Week #" << i+1 << endl;
		for (int j = 0; j < 7; j++) //days
		{
			cin >> rates[i][j];
			sum = sum + rates[i][j];
		}
		avg = sum / 7.0;
		cout << "Classification for the week:" << endl;
		if (avg <= 3.0)
		{
			cout << "Week #" << i + 1 << ": Low " ;
		}
		else if ((avg > 3.0) && (avg < 6.0))
		{
			cout << "Week #" << i + 1 << ": Medium ";
		}
		else if (avg >= 6.0)
		{
			cout << "Week #" << i + 1 << ": High ";
		}
		cout << "(" << avg << ")\n\n";
	}

	
	return 0;
}

File: 3- Second Year (2022-2023)/Term 1/OOP1 (CS311)/Lab/Labs/Lab 4/Task1/src/Task1.java
/*Write a program that calculate and classify the rainfall average for 4 weeks.
- Firstly, read the rainfall rates for each day in the 4 weeks and store them in a 2-
dimentional array [4][7].
- Write a method getClassification() to get the classification. The method should receive 
the rainfall average for a week from the main, and print the classification for that week: 
High, Medium or Low; based on the received average 

Rainfall average Classification
average >= 6 High
6 > average > 3 Medium
average < = 3 Low
*/

import java.util.Scanner;
public class Task1 {

    public static void main(String[] args){
        Scanner input = new Scanner(System.in); //declaring Scanner

        int[][] rainfall = new int[4][7]; //declaring array to store rainfall


        System.out.println("Please enter the rainfall rate for 4 weeks: ");
        for(int i = 0; i < 4; i++){ //week loop
            System.out.println("\nWeek " + (i+1) + " "); 
            for(int j = 0; j < 7; j++) //day loop
            {
                System.out.println("Day " + (j+1) + " ");
                rainfall[i][j] = input.nextInt(); 
            }
        }

        //printing list
        for(int i = 0; i < 4; i++){ //week loop
            System.out.print("\nWeek " + (i+1) + ": "); 
            for(int j = 0; j < 7; j++) //day loop
            {
                System.out.print(rainfall[i][j] + "  "); 
            }
        }
        
        //getting classification 
        System.out.println("\nClassification for each week: ");
        for(int i = 0; i < 4; i++){ //week loop
            double sum = 0;
            System.out.print("Week " + (i+1) + ": ");
            for(int j = 0; j < 7; j++) //day loop
            {
                sum += rainfall[i][j]; //getting sum of each week
            }
            getClassification(sum/7); 
        }
        input.close();
    }

    static void getClassification(double avg){
        if(avg >= 6.0)
        {
            System.out.println("High (" + avg + ")");
        }
        else if(avg < 6.0 && avg > 3.0)
        {
            System.out.println("Medium (" + avg + ")");
        }        
        else if(avg <= 3.0)
        {
            System.out.println("Low (" + avg + ")");
        }
    }
}
